### ISSUE TITLE
In certain cases `pyright` will not recognise that `__`-prefixed variable should be treated as positional only.

### ISSUE THREAD

**Issue by multimeric** [microsoft/pyright#3658]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
In certain cases `pyright` will not recognise that `__`-prefixed variable should be treated as positional only.

**To Reproduce**
```python
from typing import overload, Tuple

@overload
def foo(a, __b: bool) -> Tuple[bool]:
    ...
@overload
def foo(a, __b: int) -> Tuple[int]:
    ...
def foo(a, *args):
    return args
```
Gives me:
```
  /tmp/tmp.DIKPpaRFXw/test.py:9:5 - error: Overloaded implementation is not consistent with signature of overload 1
    Type "(a: Unknown, *args: Unknown) -> tuple[Unknown, ...]" cannot be assigned to type "(a: Unknown, __b: bool) -> Tuple[bool]"
      Keyword parameter "__b" is missing in source (reportGeneralTypeIssues)
  /tmp/tmp.DIKPpaRFXw/test.py:9:5 - error: Overloaded implementation is not consistent with signature of overload 2
    Type "(a: Unknown, *args: Unknown) -> tuple[Unknown, ...]" cannot be assigned to type "(a: Unknown, __b: int) -> Tuple[int]"
      Keyword parameter "__b" is missing in source (reportGeneralTypeIssues)
2 errors, 0 warnings, 0 informations 
Completed in 0.632sec
```

**Expected behavior**
In this case there should be no error. Mypy has no issue with the above code.

**Additional context**
* pyright 1.1.257

**Comment by multimeric** [comment]
Yes, good point. The correct signature here is for `a` to be marked as positional-only as well. I need to support older Pythons, so this works for me:
```python
from typing import overload, Tuple

@overload
def foo(__a, __b: bool) -> Tuple[bool]:
    ...
@overload
def foo(__a, __b: int) -> Tuple[int]:
    ...
def foo(__a, *args):
    return args
```
